### ISSUE TITLE
Remove dependency on IHostingEnvironment

### ISSUE THREAD

**Issue by phillijw** [severisv/SlackLogger#5]
The library comes from AspNetCore and I think it could be replaced by [Options](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/index?view=aspnetcore-2.2#using-options-and-configuration-objects) instead.

This would remove an unnecessary dependency on the AspNetCore library.

**Comment by severisv** [comment]
Seems like a good idea. Currently `IHostingEnvironment` is only used to print the environment and application name when not configured.

Would it be a good idea to get the default environment name using `Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT"`) and use the calling assembly name as the default application name?

**Comment by phillijw** [comment]
I think the main issue with doing that is that you're assuming the person is running aspnetcore. I am running many services which are not aspnet at all. That still might be a solution but I'm going to do a little more research to see if there's a more standard way without referencing aspnet at all.

Also, I just realized the Options link I posted originally was supposed to go here: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.0

**Comment by phillijw** [comment]
I think the solution is probably to just allow the user to set the environment name and application name via options. If user wants to use an environment variable, they can do that if they wish. Having the library do it doesn't make a lot of sense to me, especially if you use ASPNETCORE variable, which only applies to websites. This library can be used in non-website code as well I assume.

As for using the assembly name, I think this could work theoretically but with .net core 3 there are a some changes which might make this more of a headache than it should be so it might be a good idea to avoid it for now.

**Comment by severisv** [comment]
You make some good points 👍

It was suggested in https://github.com/severisv/SlackLogger/issues/7 that the name of the Webhook should be used when no name is specified. How do you feel about that?
It would solve the display name part of this issue.

As with the for the hosting environment, I agree that it's a little weird to be looking up `ASPNETCORE_ENVIRONMENT` when you're not running a web app, so the purist approach would definitely be to not do that and only use the provided value.
The contrary argument would be that many (most?) are using it from a web app, and it's pretty cool if it works with minimal configuration. It doesn't really do any harm (as far as I can see) to non-web applications.

**Comment by phillijw** [comment]
For #7, I think using the webhook name is probably fine. In my case we reuse the same webhook for different environments so we would want to override that anyway.

We are actually using SlackLogger in non-aspnet case and we had to add the IHostingEnvironment and IConfiguration to our dependency setup in order to use it which is why I originally created this issue. It's not something we use anywhere else in our code (we don't use json files for configuration), so it's sort of redundant in our case. If you want to leave it in, it's not too hard to work around but might trip up other people as well.

**Comment by wxtsxt** [comment]
We name our environments differently from IHostingEnvironment.EnvironmentName. To me not showing the environment name at all is best approach. I vote the client must set this
```
logging.AddSlack(options => {
   ...
   options.EnvironmentName = hostingContext.HostingEnvironment.EnvironmentName;
}
```
or no environment shows in Slack.
Warn: it is a breaking change.

IConfiguration is such a base class that it should stay.

**Comment by pavlovps** [comment]
Received the same issue while adding SlackLogger to .NET Core worker service. I did a workaround to mock IHostingEnvironment:
```
public class MockHostingEnvironment : IHostingEnvironment
    {
        public MockHostingEnvironment(string environmentName)
        {
            EnvironmentName = environmentName;
        }        
        public string ApplicationName { get; set; }
        public IFileProvider ContentRootFileProvider { get; set; }
        public string ContentRootPath { get; set; }
        public string EnvironmentName { get; set; }
        public IFileProvider WebRootFileProvider { get; set; }
        public string WebRootPath { get; set; }
    }
```

And added this one in Program.cs:
```
.ConfigureServices((hostContext, services) =>
                {
                    services.AddSingleton<Microsoft.AspNetCore.Hosting.IHostingEnvironment>(env =>
                        new MockHostingEnvironment(hostContext.HostingEnvironment.EnvironmentName));
```

**Comment by pavlovps** [comment]
In fact, if we look into LoggerProvider ctor then we can see that environment name from options is always overridden by property from IHostingEnvironment or by null
```
            _options.EnvironmentName = env?.EnvironmentName;
```

**Comment by severisv** [comment]
Fixed in 3.0
